src/Domain/CoordinateMaps/TimeDependent/Skew.cpp

```diff
@@ -232,18 +229,22 @@ template <typename T>
 std::array<tt::remove_cvref_wrap_t<T>, 3> Skew::get_width_deriv(
     const std::array<T, 3>& source_coords_cvref) const {
   using ResultT = tt::remove_cvref_wrap_t<T>;
-  std::array<ResultT, 3> source_coords{};
-  for (size_t i = 0; i < 3; i++) {
-    if constexpr (std::is_same_v<ResultT, DataVector>) {
-      // NOLINTBEGIN
-      gsl::at(source_coords, i)
-          .set_data_ref(make_not_null(&const_cast<DataVector&>(
-              dereference_wrapper(gsl::at(source_coords_cvref, i)))));
-      // NOLINTEND
-    } else {
-      gsl::at(source_coords, i) = gsl::at(source_coords_cvref, i);
+  // Can't do math with the cvref typed data so we convert it to the result
+  // type. If it's a DataVector then we use views. If it's doubles, just a copy
+  const std::array<ResultT, 3> source_coords = [&source_coords_cvref]() {
```

Comment:
Can you just do this:

```cpp
const auto& x = dereference_wrapper(source_coords_cvref[0]);
const auto& y = dereference_wrapper(source_coords_cvref[1]);
const auto& z = dereference_wrapper(source_coords_cvref[2]);
```

Then you don't need any of these const-casts etc.

Reply:
Edit: I think in other maps we just do math with `cref(DataVector)` and that seems to work, see Wedge.cpp

Reply:
No I want to do math with the entire `std::array`, not the individual components. However, that math won't work between a `std::array<DataVector>` and `std::array<std::reference_wrapper<const DataVector>>`. That's why I const cast.

Reply:
ok const-casts are pretty dangerous so maybe sprinkle some dereferencing in `StdArrayHelpers.hpp` instead?

Reply:
We use `const_cast` in a lot of places for this exact purpose (to set data references). Quick VSCode/git search shows 136 occurrences in 64 files